pkgs/development/python-modules/textual/default.nix

```diff
@@ -39,15 +40,13 @@ buildPythonPackage rec {
   ] ++ markdown-it-py.optional-dependencies.plugins ++ markdown-it-py.optional-dependencies.linkify;
 
   optional-dependencies = {
-    syntax = [
-      tree-sitter
-      tree-sitter-languages
-    ];
+    syntax = [ tree-sitter ] ++ lib.optionals (pythonOlder "3.12") [ tree-sitter-languages ];
```

Comment:
why limit the tree-sitter-languages package?

Reply:
Because of [this line](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/tree-sitter-languages/default.nix#L62), which prevents this package from building with versions of python newer than 3.11 if tree-sitter-languages is included as a dep in 3.12 and later.

Reply:
```suggestion
    syntax = [ tree-sitter ] ++ lib.optionals (!tree-sitter-languages.meta.broken) [ tree-sitter-languages ];
```

Reply:
Nice. Much cleaner solution.